jest-tests/test.test.js

```diff
@@ -48,130 +48,133 @@ describe("Jest test 1", () => {
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 0 },
-      endPosition: { row: 0, column: 13 },
-      startIndex: 0,
-      endIndex: 13,
-    });
-
-    assert(cursor.gotoFirstChild());
-    assertCursorState(cursor, {
-      nodeType: "binary_expression",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 0 },
-      endPosition: { row: 0, column: 13 },
-      startIndex: 0,
-      endIndex: 13,
-    });
-
-    assert(cursor.gotoFirstChild());
-    assertCursorState(cursor, {
-      nodeType: "binary_expression",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 0 },
-      endPosition: { row: 0, column: 5 },
-      startIndex: 0,
-      endIndex: 5,
-    });
-
-    assert(cursor.gotoFirstChild());
-    assertCursorState(cursor, {
-      nodeType: "identifier",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 0 },
-      endPosition: { row: 0, column: 1 },
-      startIndex: 0,
-      endIndex: 1,
-    });
-
-    assert(!cursor.gotoFirstChild());
-    assert(cursor.gotoNextSibling());
-    assertCursorState(cursor, {
-      nodeType: "*",
-      nodeIsNamed: false,
-      startPosition: { row: 0, column: 2 },
-      endPosition: { row: 0, column: 3 },
-      startIndex: 2,
-      endIndex: 3,
-    });
-
-    assert(cursor.gotoNextSibling());
-    assertCursorState(cursor, {
-      nodeType: "identifier",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 4 },
-      endPosition: { row: 0, column: 5 },
-      startIndex: 4,
-      endIndex: 5,
-    });
-
-    assert(!cursor.gotoNextSibling());
-    assert(cursor.gotoParent());
-    assertCursorState(cursor, {
-      nodeType: "binary_expression",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 0 },
-      endPosition: { row: 0, column: 5 },
-      startIndex: 0,
-      endIndex: 5,
-    });
-
-    assert(cursor.gotoNextSibling());
-    assertCursorState(cursor, {
-      nodeType: "+",
-      nodeIsNamed: false,
-      startPosition: { row: 0, column: 6 },
-      endPosition: { row: 0, column: 7 },
-      startIndex: 6,
-      endIndex: 7,
-    });
-
-    assert(cursor.gotoNextSibling());
-    assertCursorState(cursor, {
-      nodeType: "binary_expression",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 8 },
-      endPosition: { row: 0, column: 13 },
-      startIndex: 8,
-      endIndex: 13,
-    });
-
-    const childIndex = cursor.gotoFirstChildForIndex(12);
-    assertCursorState(cursor, {
-      nodeType: "identifier",
-      nodeIsNamed: true,
-      startPosition: { row: 0, column: 12 },
-      endPosition: { row: 0, column: 13 },
-      startIndex: 12,
-      endIndex: 13,
-    });
-    expect(childIndex).toBe(2);
-
-    assert(!cursor.gotoNextSibling());
-    assert(cursor.gotoParent());
-    assert(cursor.gotoParent());
-    assert(cursor.gotoParent());
-    assert(cursor.gotoParent());
-    assert(!cursor.gotoParent());
+    if (tree.rootNode) {
```

Comment:
(This is related to `TypeError: NodeClass is not a constructor` but not exactly)

The Jest tests were added specifically just to test that node-tree-sitter's JavaScript code handles bizarre results from the C++ code where everything is undefined in #164 . Instead of adding this `if (tree.rootNode)` check, you could just disable the Jest tests... Ideally someone would figure out why this is happening and how to solve it, since it obviously makes no sense that `jsParser.parse("a * b + c / d").rootNode` returns nothing

It has something to do with the way Jest imports stuff when running tests in worker processes and there's some sort of race condition when the native bindings are undefined. According to https://github.com/MaibornWolff/metric-gardener/pull/42#issuecomment-1921687978 

> this issue is triggered especially if the number of test files is higher than the number of worker processes used by jest

and setting Jest's `--maxWorkers` to 24 helps. I tried the `--maxWorkers` setting but it didn't help me reproduce the issue.

There was an upstream issue for this, which was closed as stale by a bot https://github.com/tree-sitter/node-tree-sitter/issues/53

Reply:
I'm inclined to just say we won't support Jest, this is clearly a bug on their end and if the PR that purportedly fixed it doesn't fix it, we should just revert it.